src/coap_debug.c

```diff
@@ -646,6 +659,10 @@ coap_show_pdu(coap_log_t level, const coap_pdu_t *pdu) {
   outbuflen = strlen(outbuf);
   snprintf(&outbuf[outbuflen], sizeof(outbuf)-outbuflen,  "\n");
   COAP_DO_SHOW_OUTPUT_LINE;
+
+#if COAP_CONSTRAINED_STACK
```

Comment:
Using `#if` rather than `#ifdef` could be useful here to allow switching off constrained stacks on Contiki and LwIP.
I have not checked but the following `#ifdef`s would lead to warnings about unused variables when compiling with `-DCOAP_CONSTRAINED_STACK=0`.

Reply:
Updated code as appropriate to support use of -DCOAP_CONSTRAINED_STACK=0 to switch off constrained stacks.